src/type_checkers/types.cr

```diff
@@ -219,25 +219,26 @@ module Mint
         node1 = prune(node1)
         node2 = prune(node2)
 
-        if node1.is_a?(Variable)
-          if node1 != node2
+        case
+        when node1.is_a?(Variable)
+          unless node1 == node2
```

Comment:
just an opinion but `unless` reduces understandability. The first thing I see is `==`. And then I have to invert it because there is some `unless`. I wouldn't write that. I know people write like that a lot in Ruby community but there are more communities in the world and I rarely see this.

Reply:
I tend to read it from left to right, so `unless` comes first, so that reads more naturally for me, also I like the positive affirmations (`==`) vs the negative (`!=`). And yes, it comes in part from Ruby, but Crystal **is** a Ruby-based (syntax-wise) language after all.